ops/testing.py

```diff
@@ -1174,6 +1178,29 @@ def reset_planned_units(self):
         """
         self._backend._planned_units = None
 
+    def add_network(
+            self,
+            network_info: '_NetworkDict',
+            endpoint_name: str,
+            relation_id: Optional[int] = None) -> None:
+        """Add simulated network data.
+
+        This supports testing charms that rely on the presence of networking info
+        to function. It will make the provided network_info available to the
+        testing backends network_get() method.
+
+        Args:
+            network_info: dictionary of network data as would be returned by
+                network_get()
+            endpoint_name: name of the binding to register data for
+            relation_id: optional a relation id
+        """
+        self._backend._network_map[(endpoint_name, relation_id)] = network_info
+
+    def reset_networks(self):
```

Comment:
wouldn't we want the ability to remove a single network? Why not `add_network/remove_network/clear_networks`?

Reply:
Ack, added a `remove_network()` now, if only for symmetry.

I'm leaning towards keeping `reset_networks()` instead of `clear_networks()` because we already have a `reset_planned_units()` and I'd thought to keep the verbs more uniform, but I'm ok with both really